lib/Doctrine/ORM/EntityManager.php

```diff
@@ -709,7 +709,7 @@ public function lock($entity, $lockMode, $lockVersion = null)
      *
      * @param string $entityName The name of the entity.
      *
-     * @return \Doctrine\ORM\EntityRepository The repository class.
+     * @return \Doctrine\Common\Persistence\ObjectRepository|\Doctrine\ORM\EntityRepository The repository class.
```

Comment:
Isn't `EntityRepository` the wrong return type for `getRepository()`? The user can change the default repository class in `Doctrine\ORM\Configuration::setDefaultRepositoryClassName()` and it has to be a class that implements `ObjectRepository`. So we had the wrong return type all the time for repository classes.

Is it a BC break to remove `EntityRepository` from the phpdoc return annotation when the type was wrong in the past?

Reply:
It's not a BC break to remove `EntityRepository`, but then I think it's also wrong: without adding a specific repository, the `EntityRepository` type hint is quite valid. From my personal experience I've also found most people to extend `EntityRepository` when creating their own repository classes, so the return type hint is still correct in most cases.

Reply:
`EntityRepository` implements `ObjectRepository` and `Selectable`, while a custom Repository class only needs to have `ObjectRepository`. If devs are using the `Selectable` implementations in `EntityRepository`, a custom Repo allows break compatibility. That's why I wasn't sure about BC.

Reply:
Like alcaeus says, technically it's not a BC break because it's not a contract in the code to return EntityRepository, but users are expecting it in autocompletion.

Reply:
EntityRepository should probably be an interface anyway (maybe in 3.0) and be made a parent of user repositories.

This return type looks strange but if people can register _anything_ as their base repository class, well...

Reply:
Honestly `EntityManager#getRepository()` should (eventually) die a terrible death.